### PR TITLE
memara/add-mock-root-functions

### PR DESCRIPTION
**Pull request by MazenEmara**

### CODE REVIEW

File: example/ios/Runner/Info.plist
Comment (abdelrahmanbonna) [inline example/ios/Runner/Info.plist:1]: please specify more details for these permissions to pass apple reviews

File: example/lib/main.dart [hunk 1]
```diff
@@ -22,6 +21,8 @@ class _MyAppState extends State<MyApp> {
       _isDev = false,
       _isExtStorage = false,
       _isEmulator = false;
+  bool _isMockLocationEnabled = false;
+  MethodChannel platform = const MethodChannel('security_plus');
```
Comment (abdelrahmanbonna) [inline example/lib/main.dart:25]: why using the method channel here while we could use it inside our plugin only

**Review by abdelrahmanbonna — CHANGES_REQUESTED** [review]
please contact me if more details needed my dear

File: example/lib/main.dart (hunk 1 above)
Comment (MazenEmara) [reply example/lib/main.dart:25]: "The getter 'isMockLocationEnabled' isn't defined for the type 'SecurityPlus'.
Try importing the library that defines 'isMockLocationEnabled', correcting the name to the name of an existing getter, or defining a getter or field named 'isMockLocationEnabled'." , i got this error when i did that at first and i didn't figure out why its correctly implemented and structured as the other functions

File: example/ios/Runner/Info.plist
Comment (MazenEmara) [reply example/ios/Runner/Info.plist:1]: i think these permissions are no longer needed, i was trying something on the swift part until we agreed its not needed for iOS.

File: example/lib/main.dart (hunk 1 above)
Comment (abdelrahmanbonna) [reply example/lib/main.dart:25]: Lets check that out tomorrow please

File: example/ios/Runner/Info.plist
Comment (abdelrahmanbonna) [reply example/ios/Runner/Info.plist:1]: Ok then please remove them as we should not add not used code

**Comment by abdelrahmanbonna** [comment]
Will reject this merge request to have a new one with the changes been done

**Pull request closed without merging**